### PR TITLE
Add direction field to sound block to enable microphone control

### LINKED ISSUE #277: Support for PulseAudio sources in the sound block
Hi there,

I'm trying go set up two sound blocks, one for input and one for output.
The wiki states, that the default for the `name` field is `@DEFAULT_SINK@`, so I thought using `@DEFAULT_SOURCE@` would work for the sound input block. 
However, setting `name` explicitly to either `@DEFAULT_SINK@` or `@DEFAULT_SOURCE@` results in the block showing 00% volume and trying to change it results in i3status-rust crashing with "Error in block 'Sound': volume unknown"

My sound block config:
```
[[block]]
block = "sound"
driver = "auto"
name = "@DEFAULT_SINK@"
on_click = "pactl set-sink-mute @DEFAULT_SINK@ toggle"
```

Btw: I think this `on_click` behavior could be a reasonable default, what do you think?

### PR DESCRIPTION
**Pull request by fedario**
This is the second attempt of this: #352

This pull request:

- direction field to sound block to enable microphone control
- adds new icons for microphone control
- renames the "volume_muted" icon to "crossed" to better reflect its new purpose

Edit: Closes #277.

### CODE REVIEW

File: src/blocks/sound.rs [hunk 1]
```diff
@@ -673,12 +901,19 @@ impl Block for Sound {
             if name.as_str() == self.id {
                 match e.button {
                     MouseButton::Right => self.device.toggle()?,
-                    MouseButton::Left =>
```
Comment (atheriel) [inline src/blocks/sound.rs]: What's the reason for moving away from `spawn_child_async()` here?

File: src/icons.rs [hunk 2]
```diff
@@ -20,11 +20,16 @@ lazy_static! {
         "update" => " UPD ",
         "toggle_off" => " OFF ",
         "toggle_on" => " ON ",
+        // This icon has no spaces around it because it is manually set as text. (sound.rs)
+        "crossed" => "x",
         "volume_full" => " VOL ",
         "volume_half" => " VOL ",
         "volume_empty" => " VOL ",
-        // This icon has no spaces around it because it is manually set as text. (sound.rs)
-        "volume_muted" => "MUTED",
```
Comment (atheriel) [inline src/icons.rs]: Can you explain the reason for changing this? And adding the `crossed` icon?

**Comment by atheriel** [comment]
Looks much better! I left a few comments for you.

**Comment by fedario** [comment]
Do you agree with my comments?

File: src/blocks/sound.rs (hunk 1 above)
Comment (SWW13) [reply src/blocks/sound.rs]: Can you then rebase your pull request?

File: src/blocks/sound.rs [hunk 3]
```diff
@@ -347,12 +378,24 @@ impl PulseAudioClient {
...
+                }
+                Direction::Output => {
+                    connection.context.borrow_mut().set_subscribe_callback(Some(Box::new(PulseAudioClient::subscribe_callback_sink)));
+                    connection.context.borrow_mut().subscribe(
+                        subscription_masks::SERVER |
+                        subscription_masks::SINK,
+                        |_| { }
+                    );
+                }
+            }
```
Comment (SWW13) [inline src/blocks/sound.rs]: This could be unified.

File: src/blocks/sound.rs [hunk 4]
```diff
@@ -364,21 +407,48 @@ impl PulseAudioClient {
...
+                        client.sender.send(request);
+                        Ok(())
+                    },
+                    Err(err) => {
+                        Err(BlockError(
+                            "sound".into(),
+                            format!("pulseaudio connection failed with error: {}", err),
+                        ))
+                    }
+                }
```
Comment (SWW13) [inline src/blocks/sound.rs]: This could be unified.

**Comment by fedario** [comment]
Hey @SWW13, sorry for the late reply, unfortunately I don't have much time at the moment. Still, I'd like to merge this feature somehow, would you mind helping me out?

**Comment by SWW13** [comment]
I'm also a bit short on time, maybe I find time for it in the next weeks.

**Comment by mariojackson** [comment]
Would be awesome if this feature would be implemented. I'm currently learning Rust, so I probably cannot help out myself, but maybe someone finds a little time in the near future.

**Comment by ammgws** [comment]
Anyone still working on this? @fedario ?

**Comment by fedario** [comment]
> Anyone still working on this? @fedario ?

It would be nice if someone could help me merge this, since it works so far.

**Comment by ammgws** [comment]
Seems like deduplicating the client is the only major hurdle left

**Comment by ammgws** [comment]
Superceded by #740

**Pull request closed without merging**